### PR TITLE
Increase phpstan level to 9

### PR DESCRIPTION
**Pull request by phil-davis**
mostly be better at declaring the various arrays of Promises, associated callables, exceptions etc.

### CODE REVIEW

File: lib/Promise.php [hunk 1]
```diff
@@ -188,16 +188,21 @@ public function wait()
...
             // If we got here, it means that the asynchronous operation
-            // errored. Therefore we need to throw an exception.
-            throw $this->value;
+            // errored. Therefore, we need to throw an exception.
+            if ($this->value instanceof \Throwable) {
+                throw $this->value;
+            }
+            // The state should have been REJECTED, with "value" a Throwable
+            // But "value" was not a Throwable. So throw a more general exception.
+            throw new \LogicException('The Promise was not fulfilled but no exception was specified');
```
Comment (phil-davis) [inline lib/Promise.php:197]: phpstan is confident that "value" will be a Throwable at this point. So I added an extra check, and throw a LogicException if value was not a Throwable. LogicException is already used elsewhere in this code, so I chose to throw that.

File: lib/Promise.php [hunk 2]
```diff
@@ -251,7 +256,11 @@ private function invokeCallback(Promise $subPromise, callable $callBack = null):
                 if (self::FULFILLED === $this->state) {
                     $subPromise->fulfill($this->value);
                 } else {
-                    $subPromise->reject($this->value);
+                    if ($this->value instanceof \Throwable) {
+                        $subPromise->reject($this->value);
+                    } else {
+                        throw new \LogicException('The subPromise was not fulfilled but no exception was specified');
+                    }
```
Comment (phil-davis) [inline lib/Promise.php:263]: phpstan is confident that "value" will be a Throwable at this point. So I added an extra check, and throw a LogicException if value was not a Throwable. LogicException is already used elsewhere in this code, so I chose to throw that.

File: lib/Loop/Loop.php [hunk 3]
```diff
@@ -302,7 +302,7 @@ protected function runStreams(?float $timeout): void
     /**
      * A list of timers, added by setTimeout.
      *
-     * @var array<int, mixed>
+     * @var array<int, array{0:float, 1:callable}>
```
Comment (staabm) [inline lib/Loop/Loop.php:305]: we might even define the callable signatures which can help to identify issues at the caller side (can be a separate PR)

**Review by staabm — APPROVED** [review]
lgtm

File: lib/Loop/Loop.php (hunk 3 above)
Comment (staabm) [inline lib/Loop/Loop.php:305]: another improvement can be to turn some of the properties into `@var list<..>` instead of array

Comment (phil-davis) [reply lib/Loop/Loop.php:305]: See PR #114
